adam_preprocessing/utils.py

```diff
@@ -231,6 +231,52 @@ def accuracy(output, target, topk=(1,)):
     return res
 
 
+def get_situation_accuracy(
+    output: torch.Tensor,
+    target: torch.Tensor,
+    situation_number_to_object_indices: Mapping[int, Sequence[int]],
+) -> float:
+    """
+    Computes the situation-wise acurracy@k for the specified values of k.
+
+    Parameters:
+        output:
+            Has shape `(B, L)` where B is the batch size and L is the number of labels. A row `i`
+            corresponds to the outputs for object `i` and a column `j` corresponds to the scores for
+            a particular candidate label `j`. That is, each entry gives the model's logit score for
+            applying some candidate label to a specific object.
+        target:
+            Has shape `(B,)`, where B is the batch size. Should be integer-type, with entries being
+            the discrete 0-based label for each object.
+        situation_number_to_object_indices:
+            Maps situation numbers to the indices of the objects belonging to that situation in the
+            full array of coordinates/inputs. A pair `(k, vs)` says that the objects at indices `vs`
+            in the output/target belong to situation number `k`.
+    """
+    pred = output.argmax(dim=1, keepdim=False).type_as(target)
+    # jac: Shouldn't be necessary, given we already forced pred to be .type_as(target), but it was
+    # in Sheng's accuracy code and I didn't care to experiment and find out.
+    target = target.type_as(pred)
+
+    n_situations = 0
+    n_correct = 0
+    for situation, object_indices in situation_number_to_object_indices.items():
+        n_situations += 1
+        if object_indices:
+            if max(object_indices) < pred.shape[0]:
```

Comment:
I'm not sure why this case is singled out as one to watch out for and provide a clear error message for

Reply:
I suppose it's not strictly necessary here.

Reply:
Removed the special error message.